components/system-broker/pkg/http/forward_headers.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package http
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/kyma-incubator/compass/components/system-broker/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// HeaderForwarder stores the specified request in the context so that they can later be used and forwarded to other backends
+func HeaderForwarder(forwardHeaders []string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+			ctx := r.Context()
+
+			for _, header := range forwardHeaders {
+				if value := r.Header.Get(header); value != "" {
```

Comment:
Missing unit tests for the case when none of the forward headers are found in the request